awscli/clidriver.py

```diff
@@ -380,12 +386,16 @@ def __call__(self, args, parsed_globals):
     def _create_command_table(self):
         command_table = OrderedDict()
         service_object = self._get_service_object()
+        service_model = self._get_service_model()
         for operation_object in service_object.operations:
             cli_name = xform_name(operation_object.name, '-')
             command_table[cli_name] = ServiceOperation(
                 name=cli_name,
                 parent_name=self._name,
+                session=self.session,
                 operation_object=operation_object,
+                operation_model=service_model.operation_model(
+                    operation_object.name),
```

Comment:
Is there any other way we can pull the ``name`` out (other than from operation objects)? That way it would be deleting arguments from calls and constructors when we do the final purge of operation objects.